packages/x-charts/src/internals/plugins/featurePlugins/useChartHighlight/useChartHighlight.ts

```diff
@@ -33,6 +34,12 @@ export const useChartHighlight: ChartPlugin<UseChartHighlightSignature> = ({ sto
   });
 
   const setHighlight = useEventCallback((newItem: HighlightItemData) => {
+    const prevItem = store.getSnapshot().highlight.item;
+
+    if (fastObjectShallowCompare(prevItem, newItem)) {
+      return;
+    }
+
     params.onHighlightChange?.(newItem);
```

Comment:
This should also prevent `onHighlightChange` from being called when the item would have been the same.